### PR TITLE
Wyg support for IDE

### PR DESCRIPTION
**Pull request by antfu**
I have finally come to this stage! 🤪

image

------

https://deploy-preview-537--wenyan-lang.netlify.com/ide

WIP

- I am planning to add wyg packages to the explorer for reference (maybe after examples)
- Maybe we can create a setting panel (just like help panel) and move the checkboxes inside

Would love to know what you think!

### CODE REVIEW

**Comment by LingDong-** [comment]
Very nice!

- Maybe we can have some indication (perhaps in compiled JS) about where each module is imported from, now that there are many sources

- Setting panel sound good. But since currently that aren't that many options. Maybe leaving them outside is ok and in fact more discoverable. But I also think that a setting panel eventually will be needed, so might as well have them now. All depends on what you would like to work on :)

Looking forward to it!

**Comment by antfu** [comment]
> Maybe we can have some indication (perhaps in compiled JS) about where each module is imported from, now that there are many sources

Sounds great. I think I can add that on next module wrapper refactor (to support py and rb as well)

> Setting panel sound good. But since currently that aren't that many options. Maybe leaving them outside is ok and in fact more discoverable. But I also think that a setting panel eventually will be needed, so might as well have them now. All depends on what you would like to work on :)

You are right. Let's keep them outside for now. I have restyled the default components to match the overall style better.

image
image
image

-----------

I think it's ready.

image

An information panel will show up when clicking the packages in the explorer.

image

File: static/ide.html [hunk 1]
```diff
@@ -44,15 +44,20 @@
...
           </ul>
         </li>
         <li>
-          <span class="caret">Examples</span>
+          <div class="caret">Examples</div>
           <ul class="nested active" id="explorer-list-examples">
           </ul>
         </li>
+        <li id="explorer-packages">
+          <div class="caret">文渊阁 Packages</div>
```
Comment (statementreply) [inline static/ide.html]: s/文渊阁/文淵閣/g

Comment (antfu) [reply static/ide.html]: Good call. Thanks 😅

**Comment by LingDong-** [comment]
Awesome! Maybe in the future when there are way too many 文淵閣 packages we need to hide them better in the GUI, but currently there are not enough so it makes sense to advertise them like right now, so great!
Thanks for the good work!

**Comment by antfu** [comment]
Yes indeed. With the packages grows, we may consider adding a packages explorer with searching. Users can add packages they want to the explorer. And yes, that's for later. 👍

**Pull request merged**